content/techniques/http-module.md

```diff
@@ -147,3 +147,29 @@ export class CatsService {
   }
 }
 ```
+
+#### Full Example
+
+Since the return of `HttpService` is an Observable, we can use `rxjs` `firstValueFrom` or `lastValueFrom` to retrieve the data of the request for us.
+
+```typescript
+import { catchError, firstValueFrom } from 'rxjs';
+
+@Injectable()
+export class CatsService {
+  private readonly logger = new Logger(CatsService.name);
+  constructor(private readonly httpService: HttpService) {}
+
+  findAll(): Promise<Cat[]> {
+    const { data } = await firstValueFrom(
+      this.httpService.get('http://localhost:3000/cats').pipe(
+        catchError((error: AxiosError) => {
+          this.logger.error(error.response.data);
+          throw 'An error happened!';
+        }),
+      ),
+    );
+    return data;
+  }
+}
+```
```

Comment:
```suggestion
`` `

> info **Hint** Visit RxJS's documentation on [`firstValueFrom`](https://rxjs.dev/api/index/function/firstValueFrom) and [`lastValueFrom`](https://rxjs.dev/api/index/function/lastValueFrom) for differences between the two
```

Don't keep the space above, I just had to add it so GitHub wouldn't change the formatting